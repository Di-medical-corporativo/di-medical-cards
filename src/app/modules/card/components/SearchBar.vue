<template>
  <div
    class="search__bar"
  >
    <input
        type="text"
        class="search__bar__input"
        placeholder="Buscar la Ficha técnica por código del producto"
        v-model="codeToSearch"
        @keyup.enter="searchTechnicalSheetByCode"
    >
    <button
        class="search__bar__button"
        :style="dimedicalBackgroundforComponent"
        @click="searchTechnicalSheetByCode"
    >
        <i class="bi bi-search icon"></i>
    </button>
  </div>
</template>

<script>
import { ref } from 'vue'
import { useRoute } from 'vue-router'
import useDimedical from '../composables/useDimedical'

export default {
  setup (_, { emit }) {
    const route = useRoute()
    const { dimedicalBackgroundforComponent } = useDimedical(route.params.sucursal)
    const codeToSearch = ref('')

    const searchTechnicalSheetByCode = () => {
      if (!codeToSearch.value || codeToSearch.value.trim().length === 0) return
      emit('code-to-search', codeToSearch.value)
    }

    return {
      dimedicalBackgroundforComponent,
      codeToSearch,
      searchTechnicalSheetByCode
    }
  }
}
</script>
