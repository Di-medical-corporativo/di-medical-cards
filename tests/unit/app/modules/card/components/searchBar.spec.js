import { shallowMount } from '@vue/test-utils'
import SearchBar from '../../../../../../src/app/modules/card/components/SearchBar.vue'

jest.mock('vue-router', () => ({
  useRoute: jest.fn(() => ({
    params: {
      id: 'kevin-aron-tapia-cruz',
      sucursal: 'sur'
    }
  })),
  useRouter: jest.fn(() => ({
    push: jest.fn()
  }))
}))

describe('Search bar component', () => {
  test('should match snapshot', () => {
    const wrapper = shallowMount(SearchBar)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should call searchTechnicalSheetByCode when button is pushed', async () => {
    const wrapper = shallowMount(SearchBar)
    const button = wrapper.find('.search__bar__button')
    const spy = (wrapper.vm.searchTechnicalSheetByCode = jest.fn())
    await button.trigger('click')
    expect(spy).toHaveBeenCalled()
  })

  test('should emit value when input has a value and button is pushed', async () => {
    const wrapper = shallowMount(SearchBar)
    const button = wrapper.find('.search__bar__button')
    wrapper.find('.search__bar__input').setValue('hola')
    await button.trigger('click')
    expect(wrapper.emitted('codeToSearch')).toBeDefined()
  })

  test('should not emit anything when input has no value', async () => {
    const wrapper = shallowMount(SearchBar)
    const button = wrapper.find('.search__bar__button')
    await button.trigger('click')
    expect(wrapper.emitted('codeToSearch')).toBeUndefined()
  })
})
