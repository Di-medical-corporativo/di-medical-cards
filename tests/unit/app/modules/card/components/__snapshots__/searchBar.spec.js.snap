// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component should match snapshot 1`] = `<div class="search__bar"><input type="text" class="search__bar__input" placeholder="Buscar la Ficha técnica por código del producto"><button class="search__bar__button" style="background: rgb(33, 141, 155);"><i class="bi bi-search icon"></i></button></div>`;
